programs/quarter_board/communication.c

```diff
@@ -0,0 +1,163 @@
+#include "ch.h"
+#include "hal.h"
+
+#include <icu_lld.h>
+#include "chprintf.h"
+#include <inttypes.h>
+
+#include "communication.h"
+
+thread_t *communicationp;
+
+MESSAGE message;
+MESSAGE *msgp = &message;
+
+uint8_t value_to_char (uint8_t character) {
+    if(character > 9) {
+       character = character - 10 + 'A';
+    } else {
+       character = character + '0';
+    }
+    return character;
+}
+
+
+uint8_t char_to_value(uint8_t character) {
+    if(character >= 'A') {

```

Comment:
When I call this function in code I am guaranteed to work with ASCII-like character set where this works.